### PR TITLE
[WIP] 📖  Add initial guide to deploying addons with tilt

### PR DESCRIPTION
**Pull request by killianmuldoon**
Signed-off-by: killianmuldoon <kmuldoon@vmware.com>

Add information to the book about how users can deploy addons with tilt, and how to develop Runtime Extensions using the tilt environment.

### CODE REVIEW

**Comment by fabriziopandini** [comment]
@killianmuldoon I’m wondering if we should move away from using addon, which is an overloaded term, and use resource instead, which is the term used by tilt or be explicit and use extensions, which is the use case we care about.
Wdyt? @sbueringer @ykakarap opinions?

**Comment by sbueringer** [comment]
> @killianmuldoon I’m wondering if we should move away from using addon, which is an overloaded term, and use resource instead, which is the term used by tilt or be explicit and use extensions, which is the use case we care about. Wdyt? @sbueringer @ykakarap opinions?

Sounds good to me. Given that the implementation doesn't really care about what it's deploying maybe something more generic like "deployment" would be slightly better than "extension"

File: docs/book/src/developer/tilt.md [hunk 1]
```diff
@@ -378,6 +378,55 @@ Set to `false` if your provider does not have a ./config folder or you do not wa
 in tilt version >= v0.22.2 (see https://blog.tilt.dev/2021/08/09/resource-grouping.html); as a convention,
 provider abbreviation should be used (CAPD, KCP etc.).
```
Comment (sbueringer) [inline docs/book/src/developer/tilt.md:379]: (unrelated to this filed just had to add it somewhere)

We're adding a Tips & Tricks section to the implementation doc in https://github.com/kubernetes-sigs/cluster-api/pull/7270.

There was a suggestion to also mention the tilt setup there:
> What if we add a note about being possible to develop and test with tilt (assuming that https://github.com/kubernetes-sigs/cluster-api/pull/7154 merge soon, this note could close the circle)
> https://github.com/kubernetes-sigs/cluster-api/pull/7270#discussion_r977565818

**Comment by killianmuldoon** [comment]
/close
Closing for now.

This needs a complete rewrite as the dev process will change in https://github.com/kubernetes-sigs/cluster-api/pull/7288

**Pull request closed without merging**